dev/com.ibm.ws.cdi.interfaces/src/io/openliberty/cdi/spi/CDIExtensionMetadata.java

```diff
@@ -69,6 +71,15 @@
         return Collections.emptySet();
     }
 
+    /**
+     * All classes returned by this method will be will be treated as bean defining annotations when CDI
+     * performs annotation scanning during application startup.
+     * All classes must be in the same archive as your CDIExtensionMetadata.
+     */
+    default public Set<Class<? extends Annotation>> getBeanDefiningAnnotationClasses() {
```

Comment:
More of a question than a comment, but the original WebSphereCDIExtension declared BDA classes using a String (i.e. "jakarta.ws.rs.Path", "jakarta.ws.rs.Provider", "jakarta.otherspec.OtherAnno"). If one of the BDA classes was not loadable (i.e. the "otherspec" feature was not enabled) the other BDA classes would still work.  

With this approach, an implementation of this interface could load the classes themselves and ignore CNFEs like:
```
@Override
public Set<Class<? extends Annotation>> getBeanDefiningAnnotationClasses() {
    Set<Class<? extends Annotation>> bdaClasses = new HashSet<>();
    for (String className : Arrays.asList("jakarta.ws.rs.Path", "jakarta.ws.rs.Provider", "jakarta.otherspec.OtherAnno")) {
        try {
            bdaClasses.add(Thread.current().getContextClassLoader().loadClass(className));
        } catch (ClassNotFoundException cnfe) {
            // ignore this one...
        }
    }
}
```

So the question is more about the dynamic nature of Liberty.  If we the "otherspec" libraries became available (either a new feature was added or the user re-deployed their app with new libraries, etc.) after CDI invoked this method, it would make no difference here, because the implementation of this interface already ignored it.  But was that the same behavior as before?  Or would the `getBeanDefiningClasses()` method be re-invoked on feature/app changes?

Reply:
Anything which causes the app to restart will result in getBeanDefiningClasses() being reinvoked.